ocaml/typing/typedecl.ml

```diff
@@ -361,16 +362,12 @@ let make_params env path params =
     List.map make_param params
 
 
-let transl_global_flags loc attrs =
-  let transl_global_flag loc (r : (bool,unit) result) =
-    match r with
-    | Ok b -> b
-    | Error () -> raise(Error(loc, Local_not_enabled))
-  in
-  let global = transl_global_flag loc (Builtin_attributes.has_global attrs) in
-  match global with
-  | true -> Types.Global
-  | false -> Types.Unrestricted
+let transl_global_flags arg =
+  match Jane_syntax.Constructor_argument.of_ast arg with
+  | Some (Jcarg_local larg, attrs) -> begin match larg with
+    | Lcarg_global arg -> Types.Global, {arg with ptyp_attributes = attrs}
```

Comment:
I understand why it's ok in the specific context of `local_` in types, but in general it's very surprising to see attributes being moved from `of_ast`'s argument to its return value. (You have to have quite a bit of context about Jane Syntax in your head to convince yourself this is ok, and this code appears outside of that context.)

Can we take advantage of any property of constructor arguments to simplify this arrangement? AFAICT, if you have a constructor declaration `type t = T of global_ int`, you can only place attributes on the `int` and not the `global_ int`. Does that suggest that Jane Syntax should handle constructor argument attributes opaquely, by transferring them to the correct syntax node, rather than passing the attributes through to the caller?

Reply:
Talked offline; we concluded that we could indeed simplify constructor arguments, but found that this could not be so nicely avoided with types; this in fact uncovered a related bug in `get_alloc_mode` (see that conversation).